File: app/src/main/java/com/example/techfix/data/dao/RepairServiceDao.java
package com.example.techfix.data.dao;
import androidx.room.*;
import com.example.techfix.data.entity.RepairService;
import java.util.List;
@Dao public interface RepairServiceDao {
 @Insert(onConflict=OnConflictStrategy.REPLACE) void insertService(RepairService service);
 @Query("SELECT * FROM repair_services") List<RepairService> getAllServices();
 @Query("SELECT * FROM repair_services WHERE categoryId = :categoryId") List<RepairService> getServicesByCategory(int categoryId);
 @Query("SELECT * FROM repair_services WHERE serviceId = :id") RepairService getServiceById(int id);
}

File: app/src/main/java/com/example/techfix/data/dao/RepairServiceDao.kt
package com.example.techfix.data.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.techfix.data.entity.RepairService

@Dao
interface RepairServiceDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertService(service: RepairService)

    @Query("SELECT * FROM repair_services")
    suspend fun getAllServices(): List<RepairService>

    @Query("SELECT * FROM repair_services WHERE categoryId = :categoryId")
    suspend fun getServicesByCategory(categoryId: Int): List<RepairService>

    @Query("SELECT * FROM repair_services WHERE serviceId = :id")
    suspend fun getServiceById(id: Int): RepairService?
}